File: 0944-Delete-Columns-to-Make-Sorted.ts
function minDeletionSize(strs: string[]): number {
  let count: number = 0;
  let m = strs.length;
  let n = strs[0].length;

  for (let j = 0; j < n; j++) {
    for (let i = 0; i < m - 1; i++) {
      if (strs[i][j].charAt(0) > strs[i + 1][j].charAt(0)) {
        count++;
        break;
      }
    }
  }

  return count;
}

File: 0944-Delete-Columns-to-Make-Sorted.go
func minDeletionSize(strs []string) int {
	if len(strs) <= 1 {
		return 0
	}
	count := 0
	m := len(strs)
	n := len(strs[0])
	for j := 0; j < n; j++ {
		for i := 0; i < m-1; i++ {
			if strs[i][j] > strs[i+1][j] {
				count++
				break
			}
		}
	}
	return count
}

